src/main/java/com/base64/d/StudyGuideChecker.java

```diff
@@ -8,127 +8,120 @@
+                // Grab the iframe
+                Element iframe = wraperDocument.select("iframe").first();
 
-//            get iframe scr url
+                // Get iframe src url
                 iframeSrc = iframe.attr("src");
 
-//            connect to the iframe source url
-                Document doc = Jsoup.connect(iframeSrc).get();
+                // Connect to the iframe source url
+                Document studyGuideDocument = Jsoup.connect(iframeSrc).get();
 
-                titleOfStudyGuide = doc3.title();
+                titleOfStudyGuide = wraperDocument.title();
                 System.out.println(titleOfStudyGuide + " " + iframeSrc);
 
-//            get the script tag info
-                Elements scriptTag = doc.getElementsByTag("script");
+                // Get the script tag info
+                Elements scriptTag = studyGuideDocument.getElementsByTag("script");
                 String jsScripts = scriptTag.toString();
 
-//            match any string of characters over 100 in length.
-                String patternString = "(\\w{100,}.+)";
-
-                Pattern pattern = Pattern.compile(patternString);
+                // Retrieve the `window.courseData` JavaScript variable.
+                // This value will be enclosed in double quotes.
+                Pattern courseDataPattern = Pattern.compile("window.courseData = \"(.*)\"");
 
-                Matcher matcher = pattern.matcher(jsScripts);
-//            System.out.println(matcher);
-                String foundBase64String = "";
-                while (matcher.find()) {
+                Matcher courseDataMatcher = courseDataPattern.matcher(jsScripts);
+                String base64String = "";
+                while (courseDataMatcher.find()) {
                     System.out.println("--------");
 
-                    foundBase64String = matcher.group(1);
+                    base64String = courseDataMatcher.group(1);
                 }
-                //reduced the giant string by 2 because it included the " and the ;
-                String trimFoundBase64String = foundBase64String.substring(0, (foundBase64String.length() - 2));
 
-                //put in this try block because one study guide had different script tag layout and threw error.
+                // Put in this try block because one study guide had different script tag layout and threw error.
                 try {
-                    // decoding a base64 string and putting into a byte array
-                    byte[] decodedArr = Base64.getDecoder().decode(trimFoundBase64String);
+                    // Decoding a base64 string and putting into a byte array
+                    byte[] decodedArr = Base64.getDecoder().decode(base64String);
                     decodedText = new String(decodedArr, "UTF-8");
-                    //System.out.println("decoded base64: " + decodedText);
-                } catch
-                (IllegalArgumentException exc) {
```

Comment:
This might have just been a typo, but I moved everything for the `catch` statement onto the same line. Splitting it up like this is a good technique if you are trying to keep your code width to a specific value (typically 80 or 120 characters).